File: 프로그래머스/lv2/42577. 전화번호 목록/전화번호 목록.java
import java.util.*;

class Solution {
    public boolean solution(String[] phoneBook) {
        Arrays.sort(phoneBook);

        for (int i = 0; i < phoneBook.length - 1; i++) {
            int n = phoneBook[i].length();
            if (phoneBook[i+1].length() >= n && phoneBook[i + 1].substring(0, n).equals(phoneBook[i])) {
                return false;
            }
        }
        return true;
    }
}

File: 프로그래머스/lv2/42577. 전화번호 목록/전화번호 목록.py
def solution(phone_book):
    
    # 1. 오름차순으로 정렬한다.
    phone_book.sort()
    
    for i in range(len(phone_book) - 1):
        n = len(phone_book[i])
        if phone_book[i+1][:n] == phone_book[i]:
            return False
                
    return True